src/internal/storage/fileset/writer.go

```diff
@@ -138,6 +138,9 @@ func (w *Writer) Copy(file File, datum string) error {
 				Datum: datum,
 			},
 		}
+		if _, ok := file.(*FileReader); ok {
+			return w.uploader.CopyByReference(copyIdx, idx.File.DataRefs)
```

Comment:
I like having CopyByReference available to us here.  It definitely simplifies things.  I don't think it's safe enough to be public though, see my other comment above.

Reply:
Refer to the relevant thread above.